### PR TITLE
Fixes #5703

### LINKED ISSUE #5703: BUG: the .sample.env should have DISABLE_LIVE_DEPLOYER=false
**Describe the bug**

The .sample.env in packages/bedrock-contracts/ folder should have DISABLE_LIVE_DEPLOYER=false instead of true. This is causing the deploy contracts script to get stuck.

**To Reproduce**
In https://stack.optimism.io/docs/build/getting-started/#deploy-the-l1-contracts, trying the "npx hardhat deploy --network getting-started --tags l1" command get stuck.

**Expected behavior**
 The script should finish deploying the contracts and run the l1 rpc.

**Screenshots**
Screenshot 2023-05-13 at 17 38 52

**System Specs:**
 - OS: MacOs
 - Package Version (or commit hash): 4611198bf8bfd16563cc6bdf49bb35eed2e46801

**Additional context**

### PR DESCRIPTION
**Pull request by chrisciszak**
**Description**

Fixes #5703

process.env will always output strings, !! in typescript will convert any non empty string to true. Adding a proper check for this resolves this problem. Also, changed the env.sample to reflect the already existing boolean logic which is easier to maintain.

Another issue that was causing the hang was related to the difference in the case that is possible to arise.

- [ ] Author or reviewer has added an entry to the [current release notes draft][RND], if appropriate.

[RND]: https://www.notion.so/oplabs/ded30107ceec41c88817e60322aa8d0a?v=b4a22cedb85a46a38c9be14e7c984953&pvs=4 "Release Notes"

### CODE REVIEW

File: packages/contracts-bedrock/deploy/019-SystemDictatorInit.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ const deployFn: DeployFunction = async (hre) => {
   if (
     (await SystemDictatorProxy.callStatic.admin({
       from: ethers.constants.AddressZero,
-    })) !== hre.deployConfig.controller
```
Comment (tynes) [inline packages/contracts-bedrock/deploy/019-SystemDictatorInit.ts:179]: This is an unrelated change, please remove it and propose in its own PR

File: packages/contracts-bedrock/src/deploy-utils.ts [hunk 2]
```diff
@@ -366,7 +366,7 @@ export const liveDeployer = async (opts: {
   hre: HardhatRuntimeEnvironment
   disabled: string | undefined
 }): Promise<boolean> => {
-  if (!!opts.disabled) {
```
Comment (tynes) [inline packages/contracts-bedrock/src/deploy-utils.ts:369]: Perhaps parse the env var into a bool higher up the call stack and have `disabled` be a bool arg

Comment (chrisciszak) [reply packages/contracts-bedrock/src/deploy-utils.ts:369]: @tynes done

File: packages/contracts-bedrock/deploy/019-SystemDictatorInit.ts (hunk 1 above)
Comment (chrisciszak) [reply packages/contracts-bedrock/deploy/019-SystemDictatorInit.ts]: moved to a new PR https://github.com/ethereum-optimism/optimism/pull/5993

**Comment by upnodedev** [comment]
We have tried your fix and it still doesn't work until I comment these lines out

image

Credit: https://github.com/ethereum-optimism/optimism/issues/5703#issuecomment-1556188148

**Comment by chrisciszak** [comment]
The fix works if you run initial deployment.

You shouldn't need to comment this.

If it failed already, then you need this too: https://github.com/ethereum-optimism/optimism/pull/5993
